$OpenBSD: patch-src_ai_lua_core_cpp,v 1.1 2018/07/29 12:57:50 kirby Exp $

backport CVE-2018-1999023 fix
upstream commit 2554c166dd45bfdee8ee24431224bd25e52bab12

Index: src/ai/lua/core.cpp
--- src/ai/lua/core.cpp.orig
+++ src/ai/lua/core.cpp
@@ -913,7 +913,7 @@ static void generate_and_push_ai_table(lua_State* L, a
 
 lua_ai_context* lua_ai_context::create(lua_State *L, char const *code, ai::engine_lua *engine)
 {
-	int res_ai = luaL_loadstring(L, code);//stack size is now 1 [ -1: ai_context]
+	int res_ai = luaL_loadbufferx(L, code, strlen(code), /*name*/ code, "t"); // [-1: AI code]
 	if (res_ai)
 	{
 
@@ -943,7 +943,7 @@ lua_ai_context* lua_ai_context::create(lua_State *L, c
 
 lua_ai_action_handler* lua_ai_action_handler::create(lua_State *L, char const *code, lua_ai_context &context)
 {
-	int res = luaL_loadstring(L, code);//stack size is now 1 [ -1: f]
+	int res = luaL_loadbufferx(L, code, strlen(code), /*name*/ code, "t");//stack size is now 1 [ -1: f]
 	if (res)
 	{
 		char const *m = lua_tostring(L, -1);
